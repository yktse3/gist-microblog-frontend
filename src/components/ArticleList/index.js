import React from 'react';
import {
  shape,
  int,
  arrayOf,
  string,
  func,
} from 'prop-types';
import Article from 'components/Article';
import { noop } from 'redux-saga/utils';
import {
  ArticleContainer,
} from './styles';

const ArticleList = ({
  articles,
  onCommentClicked,
  comments,
}) => (
  articles.map((article) => {
    return (
      <ArticleContainer
        key={`${article.id}_container`}
        id={new Date().getTime()}
      >
        <Article
          title={article.title}
          dateTime={article.created_at}
          content={article.content}
          numOfComment={article.comments}
          key={article.id}
          articleID={article.id}
          commentURL={article.comments_url}
          onCommentClicked={onCommentClicked}
          comments={comments[article.id]}
        />
      </ArticleContainer>
    );
  })
);

ArticleList.propTypes = {
  articles: arrayOf(
    shape({
      id: string,
      title: string,
      comments: int,
      comments_url: string,
      created_at: string,
      content: string,
    }),
  ),
  onCommentClicked: func,
  comments: shape(
    arrayOf(
      shape({
        id: int,
        userID: string,
        content: string,
      }),
    ),
  ).isRequired,
};

ArticleList.defaultProps = {
  articles: [],
  onCommentClicked: noop,
};

export default ArticleList;
