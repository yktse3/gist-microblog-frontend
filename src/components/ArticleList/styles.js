import styled from 'styled-components';

export const ArticleContainer = styled.div`
  margin: 20px 50px;
`;
