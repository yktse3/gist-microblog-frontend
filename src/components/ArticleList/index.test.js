import React from 'react';
import renderer from 'react-test-renderer';
import ArticleList from './index';

describe('ArticleList', () => {
  it('render successfully', () => {
    const component = renderer.create(
      <ArticleList
        articles={[
          {
            title: 'A title',
            created_at: '2018/10/15 17:13:00',
            content: 'Content',
            comments: 1,
            id: '1',
            commentURL: 'http://whatever.com/comment/1',
          },
          {
            title: 'A title 2',
            created_at: '2018/10/15 17:12:00',
            content: 'ContentContentContentContent',
            comments: 0,
            id: '2',
            commentURL: 'http://whatever.com/comment/2',
          },
        ]
        }
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
