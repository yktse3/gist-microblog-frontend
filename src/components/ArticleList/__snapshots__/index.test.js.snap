// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList render successfully 1`] = `
Array [
  <div
    className="sc-bZQynM kaOVQO"
  >
    <div
      className="sc-bxivhb kjsGUP"
    >
      <div
        className="sc-bdVaJa bPGLpq"
      >
        <span
          className="sc-bwzfXH dNCrpK"
        >
          A title
        </span>
        <span
          className="sc-htpNat cUROdq"
        >
          2018/10/15 17:13:00
        </span>
      </div>
      <span
        className="sc-ifAKCX gdOviX"
      >
        Content
      </span>
      <span
        className="sc-EHOje OnLaj"
      >
        1 comments
      </span>
    </div>
  </div>,
  <div
    className="sc-bZQynM kaOVQO"
  >
    <div
      className="sc-bxivhb kjsGUP"
    >
      <div
        className="sc-bdVaJa bPGLpq"
      >
        <span
          className="sc-bwzfXH dNCrpK"
        >
          A title 2
        </span>
        <span
          className="sc-htpNat cUROdq"
        >
          2018/10/15 17:12:00
        </span>
      </div>
      <span
        className="sc-ifAKCX gdOviX"
      >
        ContentContentContentContent
      </span>
      <span
        className="sc-EHOje OnLaj"
      >
        0 comments
      </span>
    </div>
  </div>,
]
`;
